app-layout-addon/src/main/java/com/github/appreciated/app/layout/behaviour/AbstractLeftAppLayoutBase.java

```diff
@@ -98,10 +98,11 @@ public void setIconComponent(Component appBarIconComponent) {
 
     @Override
     public void setAppLayoutContent(HasElement content) {
+        this.content.getElement().removeAllChildren();
         if (content != null) {
             this.content.getElement().appendChild(content.getElement());
-            setUpBackNavigation(content);
         }
+        setUpBackNavigation(content);
```

Comment:
If no content is provided, why bothering with setting up back navigation?

Reply:
`setUpBackNavigation` would call `paperIconButton.setIcon("menu")` in this case, since the new content is not `instanceof HasBackNavigation`.
Otherwise, the it would leave the "back"-icon.

Reply:
Not sure if I still prefer a throwing an exception instead, since setting null as content should be no option.

Reply:
This would somehow break the current behaviour, since there is a "null"-check.
So in the current version, if you set `null` as content, it would just leave the `paperIconButton` as is.

Reply:
True it changes some behaviour, ehich should not matter, only if some users decided to use this  API directly, support of this is nowhere documented. Changing this should not be critical.  
The Components of this addon do not call this method directly only indirectly via `UI.navigate()` which does not allow passing `null` afaik 🤔.

Reply:
What about "resetting" the content? In that case `null` would make sense.
Should that be supported?

Reply:
> What about "resetting" the content? In that case `null` would make sense.
Should that be supported?

There is no need to but I guess, it does not really matter 😄.